### PR TITLE
Rename metric instrument "Handles" to "Bound Instruments"

### PR DESCRIPTION
**Pull request by jmacd**
The topic of "Handle" being too general purpose has come up repeatedly.  Each time this proposed renaming is discussed, people seem to agree it's an improvement.

### CODE REVIEW

File: text/0070-metric-bound-instrument.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+method become `Bind`.
+
+Note that the phrase "bound instrument" may not appear directly in the
+user-facing API, nor is it required to, whereas the method `GetHandle`
+is a specified method on metric instruments.
+
+The newly-named `Bind()` method returns a bound instrument type.  The
+name of the returned type may simply take the name of its instrument
+with the prefix `Bound`.  For example, an `Int64Counter` instrument's
+`Bind()` method should return a `BoundInt64Counter` type.
```
Comment (bogdandrutu) [inline text/0070-metric-bound-instrument.md]: I don't really like the `BoundInt64Counter`. I would like to know what others think.

Comment (iredelmeier) [reply text/0070-metric-bound-instrument.md]: Definitely a mouthful, but nothing better comes to mind 🤔

Comment (jmacd) [reply text/0070-metric-bound-instrument.md]: I'm going to bet you would not prefer "Curried", which is another term used by Prometheus in this context.

File: text/0070-metric-bound-instrument.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+other potentially confusing terms (e.g., "TimeSeries", "Entry").  The
+term "Bound instrument" was initially suggested
+[here](https://github.com/open-telemetry/opentelemetry-specification/pull/299#discussion_r334211154)
+and widely accepted.
+
+## Internal details
+
+This is a simple renaming.  All uses of "handle" will be replaced by
+"bound instrument" in the specification.  All uses of the `GetHandle`
+method become `Bind`.
```
Comment (cijothomas) [inline text/0070-metric-bound-instrument.md]: How about `GetBoundInstrument()` as it makes it more clear that this method returns something. (like `GetHandle()` .

`Bind()` may be wrongly interpreted as an action or something one can use to manually bind instrument and labelset as in like `meter.Bind(labelset, instrument)`

Comment (yurishkuro) [reply text/0070-metric-bound-instrument.md]: It is an action, it creates a new object, bound instrument.

File: text/0070-metric-bound-instrument.md (hunk 1 above)
Comment (yurishkuro) [reply text/0070-metric-bound-instrument.md]: BoundInt64Instrument is a Go artifact. I expect it to show differently in other languages, eg BoundInstrument<long>, just BoundInstrument in Python/Js.

Comment (jmacd) [reply text/0070-metric-bound-instrument.md]: I took @bogdandrutu's meaning to be that he does not like the "BoundX" construct entirely. I don't have any better suggestions.

File: text/0070-metric-bound-instrument.md (hunk 2 above)
Comment (jmacd) [reply text/0070-metric-bound-instrument.md]: It is an action, but @cijothomas is pointing out that "Bind" doesn't imply that it returns something.

I could be convinced to move the `Bind()` method to the `Meter` as you have done, and to use a `GetBoundX` method on the instrument, but this method will just call `Bind()` (which will return something) and I'm not sure it's an improvement in clarity.

File: text/0070-metric-bound-instrument.md (hunk 1 above)
Comment (bogdandrutu) [reply text/0070-metric-bound-instrument.md]: I am fine to keep it as is.

**Pull request merged**